### PR TITLE
Update deferred loading language

### LINKED ISSUE #5165: Add deferred loading to web documentation
We should add documentation for using deferred imports in our web documentation. See https://github.com/flutter/flutter/issues/67303 for context.

### LINKED ISSUE #7117: Resolve contradictions about deferred loading between Dart and Flutter
### Page URL

https://docs.flutter.dev/perf/deferred-components
https://dart.dev/language/libraries#lazily-loading-a-library

### Page source

_No response_

### Describe the problem

I'm pretty sure that deferred loading is supported in Flutter when running on web and when running on Android. 

However, https://docs.flutter.dev/perf/deferred-components states:

> Note: This feature is currently only available on Android, taking advantage of Android and Google Play Stores’ [dynamic feature modules](https://developer.android.com/guide/playcore/feature-delivery) to deliver the deferred components packaged as Android modules. Deferred code does not impact other platforms, which continue to build as normal with all deferred components and assets included at initial install time.

And https://dart.dev/language/libraries#lazily-loading-a-library states:

> Only dart2js supports deferred loading. Flutter, the Dart VM, and dartdevc don’t support deferred loading. For more information, see [issue #33118](https://github.com/dart-lang/sdk/issues/33118) and [issue #27776.](https://github.com/dart-lang/sdk/issues/27776)

### Expected fix

If deferred loading is supported on both platforms, I would expect both notes to say so. If these are separate features (ie, "deferred loading" is different than "different components") than I would expect that to be better documented. Thanks!

### Additional context

_No response_

### LINKED ISSUE #8659: Add deferred loading for Web on 'Deferred components' page
### Page URL

https://docs.flutter.dev/perf/deferred-components/

### Page source

https://github.com/flutter/website/tree/main/src/perf/deferred-components.md

### Describe the problem

This page makes it seem like deferred loading is only possible for Android. But actually it's also possible for Web, just using slightly different infrastructure and naming. At a minimum, we should mention Web in here somewhere I think to spread the good news that Flutter web apps can be loaded in a modular way.

https://medium.com/flutter/optimizing-performance-in-flutter-web-apps-with-tree-shaking-and-deferred-loading-535fbe3cd674

### Expected fix

Ideally, separate out the parts of the article that describe Dart deferred imports (platform agnostic).

Then have a platform-specific section for Android and Web which explain how to set up each platform to make use of the deferred loading.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atsansone**
Update deferred loading language. 

Fixes #7117
Fixes #8659
Fixes #5165

Added @johnpryan due to his point on #5165 and @mkustermann due to his feedback on #7117.

----

Direct link: https://flutter-docs-prod--pr10083-fix-7717-yr828aok.web.app/perf/deferred-components

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
The first modified sentence feels chopped. Otherwise lgtm

File: src/perf/deferred-components.md [hunk 1]
```diff
@@ -14,34 +14,31 @@ features and assets when needed by the user.
 
 We refer to each uniquely downloadable bundle of Dart
 libraries and assets as a "deferred component".
-This is achieved by using Dart's deferred imports,
-which can be compiled into split AOT shared libraries.
+To load these, use Dart's deferred imports.
```
Comment (johnpryan) [inline src/perf/deferred-components.md:17]: We should link to the page that explains how to use deferred imports: https://dart.dev/language/libraries#lazily-loading-a-library

**Review by johnpryan — APPROVED** [review]
These changes look good to me, but I still feel that this page is lacking some context. It mostly explains how to get them working on Android, but a web developer would need to skip directly to step 2.

File: src/perf/deferred-components.md (hunk 1 above)
Comment (atsansone) [reply src/perf/deferred-components.md:17]: @johnpryan : That page also needs an update based on this issue. It doesn't note Android's capability.

**Review by mkustermann — APPROVED** [review]

**Pull request merged**